### PR TITLE
Fix decimals to show in TokenDisplay

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Fix an issue with how decimals were shown. We now show on `TokenDisplay` values with at least two decimal places AND at least two significant figures but ignoring trailing zeros.

### Other changes

N/A

### Tested

Manually and unit tests

### How others should test

Check that the balances you see during the send flow are formatted correctly.

### Related issues

- Part of #1126

### Backwards compatibility

N/A

### CODE REVIEW

**Review by jeanregisser — APPROVED** [review]
LGTM 👍

File: packages/mobile/src/components/TokenDisplay.tsx [hunk 1]
```diff
@@ -49,6 +49,16 @@ export function calculateDecimalsToShow(value: BigNumber) {
   return Math.abs(exponent) + 1
 }
 
+// Formats |value| so that it shows at least 2 significant figures and at least 2 decimal places without trailing zeros.
+export function formatValueToDisplay(value: BigNumber) {
+  let decimals = calculateDecimalsToShow(value)
+  let text = value.toFixed(decimals)
+  while (text[text.length - 1] === '0' && decimals-- > 2) {
```
Comment (dievazqu) [inline packages/mobile/src/components/TokenDisplay.tsx:56]: nit: I think that `2` is actually `DEFAULT_DISPLAY_DECIMALS`

**Review by dievazqu — APPROVED** [review]
LGTM! Good testing examples! :)

**Comment by ValoraQA** [comment]
Hi @gnardini we have verified the above issue and updated status in Gsheet 
Link : https://docs.google.com/spreadsheets/d/1UVmYHl_g7ZWKqX3zBtc5OdO9wEmdAdj4PW9zgwJxUys/edit#gid=0

**Verified on devices :** OnePlus 7T(11.0) , iPhone XR(14.6) ,Redmi Note 9(8.0), 
CC: @jeanregisser

**Comment by ValoraQA** [comment]
Hi @MuckT, We have verified the above issue in latest Android internal build V:1.22.0 and Test Flight build V: 1.22.0 and updated status in Gsheet

**Link :** https://docs.google.com/spreadsheets/d/1UVmYHl_g7ZWKqX3zBtc5OdO9wEmdAdj4PW9zgwJxUys/edit#gid=0

**Observation:**
- For Balance of $9.9, the displayed balance is $9.90, if after decimal there is one digit than a trailing zero is added and If there are two digits like $7.94, than displayed balance is $7.94 is displayed up to two decimals. Is it expected behavior.

**Attachments:** 
[Single digit after decimal.jpg](https://drive.google.com/file/d/1EZAkNghcyJg6BRg-NLxQnpdJZT6c9q5W/view?usp=sharing)
[Two digits after decimal.jpg](https://drive.google.com/file/d/1-WLV81v6Ry33pdeExXDL42-QnvLCiYNf/view?usp=sharing)


Verified on devices : Samsung Galaxy J5 Prime (8.0), iPhone 12(15), Redmi Note 8 (10.0)
CC: @gnardini

**Comment by ValoraQA** [comment]
Hi @MuckT we have verified the above issue in the latest  iOS Test Flight build V: 1.22.0 and updated status in Test Qulaity Tool:
Link : https://valora.testquality.com/project/12128/plan/22712/run/152953/result/2573238
Verified on devices : iPhone SE (13.6.1)
CC: @gnardini

**Pull request merged**